libs/category/src/effects/category.effects.ts

```diff
@@ -135,7 +160,20 @@ export class DaffCategoryEffects {
 				applied_sort_direction: action.sort.direction
       })
     )
-  )
+	)
+	
+	private toggleCategoryFilter(
+		toggledFilter: DaffCategoryFilterAction, 
+		appliedFilters: DaffCategoryFilterAction[]
+	): DaffCategoryFilterAction[] {
+		if(!appliedFilters) return [toggledFilter];
+
+		const nonMatchingFilters = appliedFilters.filter(filter => filter !== toggledFilter);
+
+		if(nonMatchingFilters.length === appliedFilters.length) nonMatchingFilters.push(toggledFilter);
+
+		return nonMatchingFilters.length ? nonMatchingFilters : null;
+	}
```

Comment:
This method is hard to understand. At the very least it needs comments explaining what is going on and what the purpose of the method is but I would rewrite it to make the code more clear. As I understand it, this method will either add or remove the toggled filter from the applied filters based on whether or not it is already applied. Assuming my understanding is correct, a clearer implementation would be:
```ts
private toggleCategoryFilter(
  toggledFilter: DaffCategoryFilterAction, 
  appliedFilters: DaffCategoryFilterAction[] = []
): DaffCategoryFilterAction[] {
  return appliedFilters.includes(toggledFilter)
    // filter is applied, remove it
    ? appliedFilters.filter(filter => filter !== toggledFilter)
    // filter is not applied, add it
    : appliedFilters.concat([toggledFilter])
}
```

Unless there is a reason you need to return `null`, I think this is better.